includes/admin/load.php

```diff
@@ -262,79 +320,45 @@ function perflab_activate_plugin() {
 		$plugin            = $plugin . '/' . $plugin_file_names[0];
 	}
 
-	if ( ! current_user_can( 'activate_plugin', $plugin ) ) {
-		wp_die( esc_html__( 'Sorry, you are not allowed to activate this plugin.', 'default' ) );
-	}
-
-	// Activate the plugin in question and return to prior screen.
-	$do_plugin_activation = activate_plugins( $plugin );
-	$referer              = wp_get_referer();
-	if ( ! is_wp_error( $do_plugin_activation ) ) {
-		$referer = add_query_arg(
-			array(
-				'activate' => true,
-			),
-			$referer
-		);
+	$result = activate_plugin( $plugin );
+	if ( is_wp_error( $result ) ) {
+		$status['errorMessage'] = $result->get_error_message();
+		wp_send_json_error( $status );
 	}
 
-	if ( wp_safe_redirect( $referer ) ) {
-		exit;
-	}
+	wp_send_json_success( $status );
 }
-add_action( 'admin_action_perflab_activate_plugin', 'perflab_activate_plugin' );
 
 /**
- * Callback function hooked to admin_action_perflab_deactivate_plugin to handle plugin deactivation.
+ * Callback function to handle admin inline style.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  */
-function perflab_deactivate_plugin() {
-	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
-		return;
-	}
-
-	// The plugin being deactivated.
-	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
-
-	check_admin_referer( "perflab_deactivate_plugin_{$plugin}" );
-
-	if ( ! current_user_can( 'deactivate_plugin', $plugin ) ) {
-		wp_die( esc_html__( 'Sorry, you are not allowed to deactivate this plugin.', 'default' ) );
+function perflab_print_features_page_style() {
+	?>
+<style type="text/css">
+	.plugin-card .name, .plugin-card .desc {
+		margin-left: 0;
 	}
-
-	// Deactivate the plugin in question and return to prior screen.
-	deactivate_plugins( $plugin );
-	$referer = add_query_arg(
-		array(
-			'deactivate' => true,
-		),
-		wp_get_referer()
-	);
-	if ( wp_safe_redirect( $referer ) ) {
-		exit;
+	.plugin-card-top {
+		min-height: auto;
 	}
+</style>
+	<?php
 }
-add_action( 'admin_action_perflab_deactivate_plugin', 'perflab_deactivate_plugin' );
 
 /**
  * Callback function hooked to admin_notices to render admin notices on the plugin's screen.
  *
  * @since 2.8.0
  */
 function perflab_plugin_admin_notices() {
-	if ( isset( $_GET['activate'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		?>
-		<div class="notice notice-success is-dismissible">
-			<p><?php esc_html_e( 'Plugin activated.', 'default' ); ?></p>
-		</div>
-		<?php
-	} elseif ( isset( $_GET['deactivate'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		?>
-		<div class="notice notice-success is-dismissible">
-			<p><?php esc_html_e( 'Plugin deactivated.', 'default' ); ?></p>
-		</div>
-		<?php
```

Comment:
See my other comment, the "activate" specific notice could be maintained if we continued using a simple admin action.

I would recommend to change the wording to use "Feature" intstead of "Plugin" here.